src/rabbit_mgmt_event_collector_utils.erl

```diff
@@ -185,9 +185,12 @@ ceil(TS, #state{interval = Interval}) ->
 
 handle_created(TName, Stats, Funs) ->
     Formatted = rabbit_mgmt_format:format(Stats, Funs),
-    ets:insert(TName, {{id(TName, Stats), create},
-                                              Formatted,
-                                              pget(name, Stats)}).
+    Id = id(TName, Stats),
+    ets:insert(TName, {{Id, create}, Formatted, pget(name, Stats)}),
+    case lists:member(TName, ?PROC_STATS_TABLES) of
+        true  -> ets:insert(rabbit_mgmt_stats_tables:key_index(TName), {Id});
```

Comment:
You could add a unit test to verify the indexes are cleared up. I had to do the same in other points of the code, as I had some leaks on the indexes that created noticeable memory increases reported by the users.